godel_surface_detection/src/detection/surface_detection.cpp

```diff
@@ -268,8 +270,109 @@ void SurfaceDetection::get_region_colored_cloud(sensor_msgs::PointCloud2& cloud_
   cloud_msg.header.frame_id = params_.frame_id;
 }
 
+
 bool SurfaceDetection::find_surfaces()
 {
+  ROS_INFO_STREAM("Find Surfaces Call");
+
+  // Reset members
+  surface_clouds_.clear();
+  mesh_markers_.markers.clear();
+  meshes_.clear();
+
+  // Ensure cloud ptr is not empty
+  if (full_cloud_ptr_->empty())
+    return false;
+
+  // Create Processing Cloud
+  pcl::PointCloud<pcl::PointXYZ>::Ptr process_cloud_ptr (new pcl::PointCloud<pcl::PointXYZ>());
+  process_cloud_ptr->header = full_cloud_ptr_->header;
+
+  // Subsample full cloud into processing cloud
+  for(const auto& pt : full_cloud_ptr_->points)
+  {
+    int q = rand() % DOWNSAMPLE_NUMBER;
```

Comment:
Since we're already in c++-11 land, I would really prefer to use the algorithms available in `<random>`. We'll get better quality random numbers and we should be able to control the random seed such that we can reproduce results more easily.